### PR TITLE
Move internal to public client for all API calls

### PR DESCRIPTION
**Pull request by kushalmalani**
## Description

> Describe the purpose of this pull request.

Move internal to public client for all API calls to allow for new RBAC launch

## 🎟 Issue(s)

Related #642 

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [x] Ran `make lint` before taking out of draft
- [x] Added/updated applicable tests
- [x] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

**Comment by sunkickr** [comment]
All the commands are working except `astro deployment create`. I am getting the following error: 
Screen Shot 2022-09-01 at 10 44 35 AM

**Comment by kushalmalani** [comment]
> All the commands are working except `astro deployment create`. I am getting the following error: Screen Shot 2022-09-01 at 10 44 35 AM

We confirmed offline, it was some other issue specific to the David's system.

File: cloud/deployment/deployment_variable_test.go [hunk 1]
```diff
@@ -12,9 +12,11 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+// var org = "test-org-id"
+// var ws = "test-ws-id"
+
```
Comment (neel-astro) [inline cloud/deployment/deployment_variable_test.go:17]:
```suggestion
```

**Review by neel-astro — APPROVED** [review]
Generally, LGTM ❤️, thanks for adding the `GetDeployment` method

**Review by neel-astro — APPROVED** [review]
LGTM 👍, unit tests seem to be failing but not sure if it is directly because of these changes 🤔

**Review by sunkickr — APPROVED** [review]

**Pull request merged**